src/commands/moderators/report.ts

```diff
@@ -18,41 +18,71 @@ export default {
     const member = message?.member
     if (!member) return
 
-    const messageId = BigInt(message.id)
+    const messageId = message.id
+    const reporterMemberId = interaction.user.id
+    const reporteeMemberId = member.id
 
-    // Count the number of times a message has been reported
-    const messageReportCount = await prisma.messageReportCount.upsert({
-      where: { messageId },
-      update: { count: { increment: 1 } },
-      create: { messageId, count: 1 },
-    })
+    // TODO: Maybe allow a reason to be specified? e.g. by using a modal
+    const reason = 'Reported via context menu'
 
-    const reporteeMemberId = BigInt(member.id)
+    // Save the report
+    try {
+      await prisma.messageReport.create({
+        data: {
+          messageId,
+          reporterId: reporterMemberId,
+          reporteeId: reporteeMemberId,
+          reason: reason,
+        },
+      })
+    } catch (error) {
+      if (
+        error instanceof Prisma.PrismaClientKnownRequestError &&
+        error.code === 'P2002'
```

Comment:
Should we use this [prisma-error-enum](https://github.com/vinpac/prisma-error-enum)? For better error code readability.